### PR TITLE
Topic/refactor config

### PR DESCRIPTION
**Pull request by ohdearaugustin**
Refactor configuration to a more default cobra config layout. Makes it more maintainable and easier extendable.

### CODE REVIEW

**Review by juanfont — APPROVED** [review]
This looks awesome. And indeed way more tidy! :)

File: cmd/headscale/cli/preauthkeys.go [hunk 1]
```diff
@@ -10,12 +10,22 @@ import (
...
 
-var PreauthkeysCmd = &cobra.Command{
+func init() {
+	rootCmd.AddCommand(preauthkeysCmd)
+	preauthkeysCmd.PersistentFlags().StringP("namespace", "n", "", "Namespace")
+	preauthkeysCmd.AddCommand(listPreAuthKeys)
+	preauthkeysCmd.AddCommand(createPreAuthKeyCmd)
+	createPreAuthKeyCmd.PersistentFlags().Bool("reusable", false, "Make the preauthkey reusable")
+	createPreAuthKeyCmd.PersistentFlags().Bool("ephemeral", false, "Preauthkey for ephemeral nodes")
+    createPreAuthKeyCmd.Flags().StringP("expiration", "e", "", "Human-readable expiration of the key (30m, 24h, 365d...)")
```
Comment (cure) [inline cmd/headscale/cli/preauthkeys.go:24]: Looks like there's a whitespace offset here that isn't quite right

**Review by cure — APPROVED** [review]
I agree, this is awesome, thanks!!

**Comment by juanfont** [comment]
@ohdearaugustin can you check the minor thing @cure pointed out? I am really looking forward merging this one :smile:

**Comment by juanfont** [comment]
Just merged. Will fix it myself :smiley:

**Comment by ohdearaugustin** [comment]
Sorry was on holidays and didn't took my laptop with me. Thank you for fixing it. (=

**Pull request merged**